File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.py
def solution(n, stations, w):
    answer = 0
    p = 0
    limit = 2*w+1
    for station in stations:
        left = station - w
        right = station + w
        
        if p < left -1:
            need = left - p - 1
            answer += (need + limit - 1) // limit
        
        p = right
        
    if p < n:
        need = n - p
        answer += (need + limit - 1) // limit
            
    return answer

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.cpp
#include <iostream>
#include <vector>
using namespace std;

int solution(int n, vector<int> stations, int w)
{
    int answer = 0;
    int range = 2 * w + 1;
    int start = 1;

    for (int station : stations) {
        int left = station - w;
        int right = station + w;
        
        if (start < left) {
            int len = left - start;
            answer += (len + range - 1) / range;
        }

        start = right + 1;
    }

    if (start <= n) {
        int len = n - start + 1;
        answer += (len + range - 1) / range;
    }

    return answer;
}
